Data/DataModel/PuzzleStatePerTeam.cs

```diff
@@ -64,5 +64,53 @@ public bool IsSolved
         /// Notes input by the team for this puzzle
         /// </summary>
         public string Notes { get; set; }
+
+        /// <summary>
+        /// If set, represents the time when a team can begin submitting answers for the
+        /// puzzle again. If a team spams answers to a puzzle, they will be locked-out
+        /// from submitting additional answers for a brief period of time.
+        /// </summary>
+        public DateTime? LockoutTime { get; set; }
+
+        /// <summary>
+        /// Returns whether or not the team is currently locked out of a puzzle.
+        /// </summary>
+        [NotMapped]
+        public bool IsTeamLockedOut
+        {
+            get { return LockoutTime == null ? false : DateTime.UtcNow.CompareTo(LockoutTime) < 0; }
+        }
+
+        /// <summary>
+        /// Returns how much lockout time is left.
+        /// </summary>
+        [NotMapped]
+        public TimeSpan? LockoutTimeRemaining
+        {
+            get { return LockoutTime == null ? null : LockoutTime - DateTime.UtcNow; }
```

Comment:
Perhaps make this a true TimeSpan and return TimeSpan.Zero, and also clipping negative times to TimeSpan.Zero